File: sdks/nodejs/src/providers/google.ts
import type { CyberArmorClient } from '../client';
import { CyberArmorOpenAICompatible } from './openaiCompatible';

export class CyberArmorGoogle extends CyberArmorOpenAICompatible {
  constructor(ca: CyberArmorClient, options?: Record<string, unknown>) {
    super(ca, 'google', options, 'gemini-2.0-flash');
  }
}

export function protectedGoogleClient(ca: CyberArmorClient, options?: Record<string, unknown>): CyberArmorGoogle {
  return new CyberArmorGoogle(ca, options);
}

File: sdks/rust/src/providers/google.rs
use std::sync::Arc;
use crate::client::CyberArmorClient;
use super::openai_compatible::OpenAICompatibleProvider;

/// Builds a Google provider wrapper backed by an OpenAI-compatible endpoint.
pub fn new(client: Arc<CyberArmorClient>, api_key: impl Into<String>, tenant_id: impl Into<String>) -> OpenAICompatibleProvider {
    OpenAICompatibleProvider::new(client, api_key, "google", tenant_id, "https://generativelanguage.googleapis.com/v1beta/openai", "gemini-2.0-flash")
}
